digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -513,6 +513,34 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
     }
   }
 
+  def createRoadLinkProject(roadAddressProject: RoadAddressProject) = {
+    withDynTransaction {
+      val id = Sequences.nextViitePrimaryKeySeqValue
+      var project = roadAddressProject.copy(id= id)
+
+      try {
+        val roadAddressExists = RoadAddressDAO.getRoadAddressProjectById(project.id) match{
```

Comment:
roadaddressExists doesnt seem to be used anywhere `RoadAddressDAO.getRoadAddressProjectById(project.id) match{`  is probably enough